File: JavaIrrgarten/irrgarten/Monster.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author aulas
 */
public class Monster {
    private static int INITIAL_HEALTH = 5;
    private String name;
    private float intelligence;
    private float strength ;
    private float health;
    private int row;
    private int col;
    private static final int FUERA_TABLERO = -1;
    
    public Monster(String name,float intelligence,float strength){
        this.name = name;
        this.intelligence = intelligence;
        this.strength = strength;
        health = INITIAL_HEALTH;
        row = FUERA_TABLERO;
        col = FUERA_TABLERO;
    }
    public boolean dead(){
        boolean vivo = true;
        if(health < 1){
            vivo = false;
        }
        return vivo;
    }
    
    public float attack(){
        return Dice.intensity(this.strength);
    }
    
    public void setPos(int row,int col){
        this.row = row;
        this.col = col;
    }
    @Override
    public String toString(){
        return "Name: " + name + " Intelligence : " + intelligence + " " +
               " Strength:" + strength + " " + 
               "Health: " + health + " " +
               "Position:" + "[" + row + "," + col + "]";
    }
    
    private void gotWounded(){
        health--;
    }
    
    public boolean defend(float receiveAttack){
        boolean isDead =dead();
        if(!isDead){
            float defensiveEnergy = Dice.intensity(intelligence);
            if(defensiveEnergy < receiveAttack){
                gotWounded();
                isDead = dead();
            }
        }
        return isDead;
        
    }
    
    
}

File: RubyIrrgarten/Irrgarten1/Monster.rb
require_relative 'Dice'

module Irrgarten
  class Monster
    @@INITIAL_HEALTH = 5
    @@OUT_LABYRINTH = -1

    def initialize(name,intelligence,strength)
      @name = name
      @intelligence = intelligence
      @strength = strength
      @health = @@INITIAL_HEALTH
      @row = @@OUT_LABYRINTH
      @col = @@OUT_LABYRINTH
    end

    def dead
      return @health < 1
    end

    def attack
      return Dice.intensity(@strength)
    end

    def defend(received_attack)
      isDead = dead
      if !isDead
        defensive_energy = Dice.intensity(@intelligence)
        if defensive_energy < received_attack
          got_wounded
          isDead = dead
        end
      end
      return isDead
    end

    def set_pos(row,col)
      @row = row
      @col = col
    end

    def to_s
      return "Name: #{@name} intelligence: #{@intelligence} strength: #{@strength} health: #{@health} position: #{@row},#{@col}"
    end

    private
    def got_wounded
      @health -= 1
    end
  end
end